src/HVGroup.cpp

```diff
@@ -7,51 +10,94 @@ HVGroup::HVGroup(Interface& m_interface):
 
         m_box = new QGroupBox("HV Settings", this);
 
-        m_label = new QLabel("HV value:");
-        
-        m_value_label = new QLabel("0");
-        m_value_label->hide();
+        m_layout = new QGridLayout();
 
-        m_spin = new QSpinBox(0);
-        m_spin->setRange(0, 2000);
-        m_spin->setSingleStep(1);
+        int temp_iterator = 0;
+        for (int setHVValue : m_interface.getConditions().getHVPMTSetValues()){
```

Comment:
Why not do something like
```
const std::vector<int>& HVMPTSetValues = m_interface.getConditions().getHVPMTSetValues();
for(size_t id = 0; id < HVMPTSetValues.size(); id++) {
    // value = HVMPTSetValues[i];
}
```
It feels more natural than iterating on the values themselves... But it's probably picketing.

Whatever you go for, please add a space between `)` and `{` ;)